src/query/interpreter.cpp

```diff
@@ -416,19 +421,30 @@ Callback HandleReplicationQuery(ReplicationQuery *repl_query, const Parameters &
   ExpressionEvaluator evaluator(&frame, symbol_table, evaluation_context, db_accessor, storage::View::OLD);
 
   Callback callback;
+  std::optional<Notification> replica_notification;
```

Comment:
By adding 
```
Notification(Notification&&) = default;
Notification& operator=(Notification&&) = default;
```
The `Notification ` should be move constructible/assigneable. In that case I think the `optional` is not necessary.